### PR TITLE
fix: recover daemon startup from corrupt state file

### PR DESCRIPTION
**Pull request by mblarsen**
## Summary
- fallback to an empty daemon state when startup cannot load state.json
- normalize daemon state in NewDaemon so nil state pointers and nil collections cannot panic runtime loops
- add regression tests for corrupt-state fallback and nil-state daemon initialization

## Validation
- go test ./...
- manual repro with malformed state file ({) now logs a warning and daemon starts without panic

### CODE REVIEW

File: cmd/daemon.go [hunk 1]
```diff
@@ -110,6 +104,17 @@ var runCmd = &cobra.Command{
 	},
 }
 
+func loadDaemonState(statePath string) *daemon.State {
+	state, err := daemon.LoadState(statePath)
+	if err != nil {
+		slog.Warn("Failed to load daemon state; starting with empty state", "path", statePath, "err", err)
+		return daemon.NewState()
+	}
```
[the remark this answers is not in the document]
Comment (mblarsen) [reply cmd/daemon.go:118]: Great catch — I tightened this so recovery only happens for malformed JSON state, not all load errors.

What changed:
- `internal/daemon/state.go`: introduced `ErrMalformedState` and `IsCorruptStateError(err)`.
- `LoadState` now wraps JSON unmarshal failures with `ErrMalformedState`.
- `cmd/daemon.go`: `loadDaemonState` now returns `(*daemon.State, error)` and only falls back to `daemon.NewState()` when `daemon.IsCorruptStateError(err)` is true.
- Non-corrupt read errors (permission/I/O, directory path, etc.) are now returned and startup fails loudly.

Why this works:
- Corrupt payloads (`state.json` parse/invalid content) still recover safely to prevent panic.
- Read/access failures no longer drop tracked leases silently; daemon exits with explicit startup error so operators can fix the underlying storage issue.

Coverage added:
- `TestLoadDaemonStateReturnsErrorForUnreadableState`
- `TestState/load corrupted` asserts `IsCorruptStateError == true`
- `TestState/load read failure is not corrupt` asserts `IsCorruptStateError == false`

Also re-ran `go test ./...` and manually validated both malformed and unreadable startup paths.

**Comment by mblarsen** [comment]
@codex I've addressed your issues please review again
